rules/TypeDeclaration/TypeAnalyzer/PropertyFetchTypeAnalyzer.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\TypeDeclaration\TypeAnalyzer;
+
+use PhpParser\Node\Expr;
+use PhpParser\Node\Expr\PropertyFetch;
+use PhpParser\Node\Identifier;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\Php\PhpPropertyReflection;
+use PHPStan\Reflection\PropertyReflection;
+use PHPStan\Reflection\WrapperPropertyReflection;
+use PHPStan\Type\MixedType;
+use Rector\NodeTypeResolver\Node\AttributeKey;
+
+final class PropertyFetchTypeAnalyzer
+{
+    public function isPropertyFetchExprNotNativelyTyped(Expr $expr): bool
+    {
+        if (! $expr instanceof PropertyFetch) {
+            return false;
+        }
+
+        if (! $expr->name instanceof Identifier) {
+            return false;
+        }
+
+        $scope = $expr->getAttribute(AttributeKey::SCOPE);
+        if (! $scope instanceof Scope) {
+            return false;
+        }
+
+        $propertyName = $expr->name->toString();
+        $propertyHolderType = $scope->getType($expr->var);
+        if (! $propertyHolderType->hasProperty($propertyName)->yes()) {
+            return false;
+        }
+
+        $originalProperty = $propertyHolderType->getProperty($propertyName, $scope);
+        $nativePropertyReflection = $this->getNativeReflectionForProperty($originalProperty);
+
+        if ($nativePropertyReflection === null) {
+            return false;
+        }
+
+        if ($nativePropertyReflection->getNativeType() instanceof MixedType) {
+            return true;
+        }
+
+        return false;
```

Comment:
direct return can be applied here, you can fix by run rector to this file:

```
bin/rector process rules/TypeDeclaration/TypeAnalyzer/PropertyFetchTypeAnalyzer.php
composer fix-cs -- rules/TypeDeclaration/TypeAnalyzer/PropertyFetchTypeAnalyzer.php
```

Reply:
Thanks for the fast feedback @samsonasik! Direct return was indeed cleaner here 😄